src/EventGrid/06-15-2022_design_review.md

```diff
@@ -0,0 +1,1067 @@
+
+    - `{ SAMPLE USAGE HERE }`
+
+## Specific test cases
+
+Please provide a link to the API tests that exist for this feature. 
+
+- [azure-sdk-for-net/ScenarioTests.PartnerTests.CRUD.cs at main · Azure/azure-sdk-for-net (github.com)](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventgrid/Microsoft.Azure.Management.EventGrid/tests/Tests/ScenarioTests.PartnerTests.CRUD.cs)
+
+## Syntax changes
+
+This should include PowerShell-help style syntax descriptions of all new and changed cmdlets, similar to the syntax portion of PowerShell help (or markdown help), for example:
+
+### New Cmdlets
+
+```powershell
+NAME
+	Get-AzureEventGridPartnerConfiguration
+
+SYNOPSIS
+	Gets a partner configuration or lists all partner configurations in the current Azure subscription.
+
+PS C:\> Get-AzureEventGridPartnerConfiguration 
+	[-ResourceGroupName] <String> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+```
+
+```powershell
+NAME
+	Get-AzureEventGridPartnerRegistration
+
+SYNOPSIS
+	Gets a partner registration or lists all partner registrations in the current Azure subscription.
+
+PS C:\> Get-AzureEventGridPartnerRegistration 
+	[-ResourceGroupName] <String> 
+	[-Name] <string> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+PS C:\> Get-AzureEventGridPartnerRegistration [-ResourceId] <String> [-ODataQuery <String>] [-Top <Int32>] [<CommonParameters>]
+
+```
+
+```powershell
+NAME
+	Get-AzureEventGridPartnerDestination
+
+SYNOPSIS
+	Gets a partner destination or lists all partner registrations in the current Azure subscription.
+
+PS C:\> Get-AzureEventGridPartnerDestination 
+	[-ResourceGroupName] <String> 
+	[-Name] <string> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+PS C:\> Get-AzureEventGridPartnerDestination 
+	[-ResourceId] <String> 
+	[-Name] <string> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+```
+
+```powershell
+NAME
+	Get-AzureEventGridVerifiedPartner
+
+SYNOPSIS
+	Gets a verified partner or lists all verified partners in the current tenant.
+
+PS C:\> Get-AzureEventGridVerifiedPartner 
+	[-VerifiedPartnerName <String>] 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+```
+
+```powershell
+NAME
+	Get-AzureEventGridPartnerNamespace
+
+SYNOPSIS
+	Gets a partner namespace or lists all partner namespaces in the current Azure subscription.
+
+PS C:\> Get-AzureEventGridPartnerNamespace 
+	[-ResourceGroupName] <String> 
+	[-Name] <String> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>]
+    [<CommonParameters>]
+
+PS C:\> Get-AzureEventGridPartnerNamespace 
+	[-ResourceId] <String> 
+	[-ODataQuery <String>] 
+	[-Top <Int32>] 
+	[<CommonParameters>]
+
+```
+
+```powershell
+NAME
+	Get-AzureEventGridChannel
+
+SYNOPSIS
+	Gets a channel or lists all channels under a partner namespace.
+
+PS C:\> Get-AzureEventGridChannel 
+	[-ResourceGroupName] <String> 
```

Comment:
ResourceGroupName not optional

Reply:
PartnerNamespaceName not optional

Reply:
I was treating it as a positional variable. The only thing that is optional here is the explicit name of the variable, the value is still required.